src/DDTrace/Integrations/Memcached/MemcachedSandboxedIntegration.php

```diff
@@ -0,0 +1,249 @@
+                return false;
+            }
+            $integration->setCommonData($span, $command);
+            if (!is_array($args[0])) {
+                $integration->setServerTagsByKey($span, $this, $args[0]);
+                $span->meta['memcached.query'] = $command . ' ' . Obfuscation::toObfuscatedString($args[0]);
+                $span->meta['memcached.server_key'] = (string)$args[0];
+            }
+
+            $integration->markForTraceAnalytics($span, $command);
+        });
+    }
+
+    public function traceMulti($command)
+    {
+        $integration = $this;
+        dd_trace_method('Memcached', $command, function (SpanData $span, $args) use ($integration, $command) {
+            if (dd_trace_tracer_is_limited()) {
+                return false;
+            }
+            $integration->setCommonData($span, $command);
+            if (!is_array($args[0])) {
+                $integration->setServerTagsByKey($span, $this, $args[0]);
+            }
+            $span->meta['memcached.query'] = $command . ' ' . Obfuscation::toObfuscatedString($args[0], ',');
+            $integration->markForTraceAnalytics($span, $command);
+        });
+    }
+
+    public function traceMultiByKey($command)
+    {
+        $integration = $this;
+        dd_trace_method('Memcached', $command, function (SpanData $span, $args) use ($integration, $command) {
+            if (dd_trace_tracer_is_limited()) {
+                return false;
+            }
+            $integration->setCommonData($span, $command);
+            $span->meta['memcached.server_key'] = (string)$args[0];
+            $integration->setServerTagsByKey($span, $this, $args[0]);
+            $query = "$command " . Obfuscation::toObfuscatedString($args[1], ',');
+            $span->meta['memcached.query'] = $query;
+            $integration->markForTraceAnalytics($span, $command);
+        });
+    }
+
+    /**
+     * Sets common values shared by many commands.
+     *
+     * @param SpanData $span
+     * @param string $command
+     */
+    public function setCommonData(SpanData $span, $command)
+    {
+        $span->name = "Memcached.$command";
+        $span->type = Type::MEMCACHED;
+        $span->service = 'memcached';
+        $span->resource = $command;
+        $span->meta['memcached.command'] = $command;
+    }
+
+    /**
+     * Memcached::getServerByKey() /might/ return incorrect information if the
+     * distribution would be rebuilt on a real call (Memcached::get(),
+     * Memcached::getByKey(), and other commands that actually hit the server
+     * include logic to regenerate the distribution if a server has been ejected
+     * or if a timer expires; Memcached::getServerByKey() does not check for the
+     * distribution being rebuilt. Getting around that would likely be
+     * prohibitively expensive though.
+     */
+    public function setServerTagsByKey(SpanData $span, $memcached, $key)
+    {
+        $server = $memcached->getServerByKey($key);
```

Comment:
Since [Memcached ::getServerByKey()](https://www.php.net/manual/en/memcached.getserverbykey.php) can return `false`, can add a check on the return value? Something like:

```php
if (!is_array($server)) {
    return;
}
```

I see this was migrated from the original integration so it would be great to update it there too. :)

Reply:
good catch 👍